src/BigNumber.php

```diff
@@ -325,11 +337,26 @@ private static function cleanUp(string $number) : string
             return '0';
         }
 
-        if ($firstChar === '-') {
-            return '-' . $number;
+        return $firstChar === '-' ? '-' . $number : $number;
+    }
+
+    /**
+     * Removes optional leading zeros and applies sign if needed(- for negatives).
+     *
+     * @param string $number The number, validated as a non-empty string of digits
+     * @param string $sign   The sign,  + or -
+     *
+     * @psalm-pure
+     */
+    private static function leanCleanUp(string $sign, string $number) : string
```

Comment:
This method can replace `cleanUp` and accept `null` for convenience:

```suggestion
    private static function cleanUp(?string $sign, string $number) : string
```

It can be used this way for numerator/denominator:

```diff
-             if ($sign !== null) {
-                 $numerator = $sign . $numerator;
-             }
-
-             $numerator   = self::cleanUp($numerator);
-             $denominator = self::cleanUp($denominator);
+
+             $numerator   = self::cleanUp($sign, $numerator);
+             $denominator = self::cleanUp($sign, $denominator);
```